src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/AssessmentService.kt

```diff
@@ -97,6 +100,29 @@ class AssessmentService(
     assessmentRepository.save(assessmentEntity)
   }
 
+  @Transactional
+  fun submitForPreDecisionChecks(prisonNumber: String) {
+    val assessmentEntity = getCurrentAssessment(prisonNumber).assessmentEntity
+    assessmentEntity.performTransition(CompleteAddressChecks)
+    assessmentRepository.save(assessmentEntity)
+  }
+
+  @Transactional
+  fun updateAddressChecksStatus(prisonNumber: String, status: ResidentialChecksStatus) {
+    val currentAssessment = getCurrentAssessment(prisonNumber)
```

Comment:
does this mean that the frontend needs to work out whether the checks are complete or not? 
how does it do that?

Reply:
ah no it's in the ResidentialChecksService